tests/includes/class-test-functions.php

```diff
@@ -216,4 +216,111 @@ public function object_to_uri_provider() {
 			),
 		);
 	}
+
+	/**
+	 * Test is_activity with array input.
+	 *
+	 * @covers ::is_activity
+	 */
+	public function test_is_activity_with_array() {
+		// Test valid activity types.
+		$valid_activities = array(
+			array( 'type' => 'Create' ),
+			array( 'type' => 'Update' ),
+			array( 'type' => 'Delete' ),
+			array( 'type' => 'Follow' ),
+			array( 'type' => 'Accept' ),
+			array( 'type' => 'Reject' ),
+			array( 'type' => 'Add' ),
+			array( 'type' => 'Remove' ),
+			array( 'type' => 'Like' ),
+			array( 'type' => 'Announce' ),
+			array( 'type' => 'Undo' ),
+		);
+
+		foreach ( $valid_activities as $activity ) {
+			$this->assertTrue(
+				\Activitypub\is_activity( $activity ),
+				sprintf( 'Activity type %s should be valid', $activity['type'] )
+			);
+		}
+
+		// Test invalid activity types.
+		$invalid_activities = array(
+			array( 'type' => 'Note' ),
+			array( 'type' => 'Article' ),
+			array( 'type' => 'Person' ),
+			array( 'type' => 'Image' ),
+			array( 'type' => 'Video' ),
+			array( 'type' => 'Audio' ),
+			array( 'type' => '' ),
+			array( 'type' => null ),
+			array(),
+		);
+
+		foreach ( $invalid_activities as $activity ) {
+			$this->assertFalse(
+				\Activitypub\is_activity( $activity ),
+				sprintf( 'Activity type %s should be invalid', isset( $activity['type'] ) ? $activity['type'] : 'null' )
+			);
+		}
```

Comment:
Let's break these out in two separate tests with `dataProvider`s for each Activity type